bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeConverter.java

```diff
@@ -463,6 +469,51 @@ private static boolean isConvertibleToArrayType(Object sourceValue, BArrayType t
         return true;
     }
 
+    private static boolean isConvertibleToTupleType(Object sourceValue, BTupleType targetType,
+                                                    List<TypeValuePair> unresolvedValues) {
+        if (!(sourceValue instanceof ArrayValue)) {
+            return false;
+        }
+
+        ArrayValue source = (ArrayValue) sourceValue;
+        List<Type> targetTypes = targetType.getTupleTypes();
+        int sourceTypeSize = source.size();
+        int targetTypeSize = targetTypes.size();
+        Type targetRestType = targetType.getRestType();
+
+        if (sourceTypeSize < targetTypeSize) {
+            return false;
+        }
+        if (targetRestType == null && sourceTypeSize > targetTypeSize) {
+            return false;
+        }
+
+        Set<Type> convertibleTypes;
+
+        for (int i = 0; i < targetTypeSize; i++) {
+            convertibleTypes = getConvertibleTypes(source.getRefValue(i), targetTypes.get(i), unresolvedValues);
+            if (convertibleTypes.isEmpty()) {
+                return false;
+            }
+            if (convertibleTypes.size() > 1 && !convertibleTypes.contains(TypeChecker.getType(source.getRefValue(i)))
+                    && !hasIntegerSubTypes(convertibleTypes)) {
+                return false;
+            }
```

Comment:
we can combine the if the two if statements using an `||` and move the boolean checks to a separate method

Reply:
The both `if` checks have been moved to a separate method and `convertibleTypes.isEmpty()` has been kept as an early separate check to prevent generating multiple instructions on this particular failing case.